src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Utility network/PerformValveIsolationTrace/PerformValveIsolationTrace.xaml.cs

```diff
@@ -0,0 +1,156 @@
+        {
+            try
+            {
+                // Disable the UI.
+                FilterOptions.Visibility = Visibility.Collapsed;
+
+                // Create and load the utility network.
+                _utilityNetwork = await UtilityNetwork.CreateAsync(new Uri(FeatureServiceUrl));
+
+                // Create a map with layers in this utility network.
+                MyMapView.Map = new Map(Basemap.CreateStreetsNightVector());
+                MyMapView.Map.OperationalLayers.Add(new FeatureLayer(new Uri($"{FeatureServiceUrl}/{LineLayerId}")));
+                MyMapView.Map.OperationalLayers.Add(new FeatureLayer(new Uri($"{FeatureServiceUrl}/{DeviceLayerId}")));
+
+                // Get a trace configuration from a tier.
+                UtilityDomainNetwork domainNetwork = _utilityNetwork.Definition.GetDomainNetwork(DomainNetworkName) ?? throw new ArgumentException(DomainNetworkName);
+                UtilityTier tier = domainNetwork.GetTier(TierName) ?? throw new ArgumentException(TierName);
+                _configuration = tier.TraceConfiguration;
+
+                // Create a trace filter.
+                _configuration.Filter = new UtilityTraceFilter();
+
+                // Get a default starting location.
+                UtilityNetworkSource networkSource = _utilityNetwork.Definition.GetNetworkSource(NetworkSourceName) ?? throw new ArgumentException(NetworkSourceName);
+                UtilityAssetGroup assetGroup = networkSource.GetAssetGroup(AssetGroupName) ?? throw new ArgumentException(AssetGroupName);
+                UtilityAssetType assetType = assetGroup.GetAssetType(AssetTypeName) ?? throw new ArgumentException(AssetTypeName);
+                Guid globalId = Guid.Parse(GlobalId);
+                _startingLocation = _utilityNetwork.CreateElement(assetType, globalId);
+
+                // Create a graphics overlay.
+                GraphicsOverlay overlay = new GraphicsOverlay();
+                MyMapView.GraphicsOverlays.Add(overlay);
+                Symbol symbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Cross, System.Drawing.Color.LimeGreen, 25d);
+
+                // Display starting location.
+                IEnumerable<ArcGISFeature> elementFeatures = await _utilityNetwork.GetFeaturesForElementsAsync(new List<UtilityElement> { _startingLocation });
+                MapPoint startingLocationGeometry = elementFeatures.FirstOrDefault().Geometry as MapPoint;
+                Graphic graphic = new Graphic(startingLocationGeometry, symbol);
+                overlay.Graphics.Add(graphic);
+
+                // Set the starting viewpoint.
+                await MyMapView.SetViewpointAsync(new Viewpoint(startingLocationGeometry, 3000));
+
+                // Build the choice list for categories populated with the `Name` property of each `UtilityCategory` in the `UtilityNetworkDefinition`.
+                Categories.ItemsSource = _utilityNetwork.Definition.Categories;
```

Comment:
I recommend setting a default selected item, otherwise there is an exception when clicking trace.

Reply:
Applies to forms as well